File: greedy/45.go
import "math"

func jump(nums []int) int {
	list := make([]int, len(nums))
	for i := 1; i < len(nums); i++ {
		list[i] = math.MaxInt32
	}
	for i := 0; i < len(nums)-1; i++ {
		for j := 1; j <= nums[i]; j++ {
			if i+j < len(nums) && list[i]+1 < list[i+j] {
				list[i+j] = list[i] + 1
			}
		}
	}
	return list[len(nums)-1]
}

File: greedy/45.cc
class Solution {
public:
    int jump(vector<int>& nums) {
        std::vector<int> dp(nums.size(), -1);
        dp[0] = 0;
        for (int i = 0; i < nums.size(); ++i) {
            if (dp[i] == -1)
                continue;
            for (int j = 0; j <= nums[i]; ++j) {
                if ((i+j) >= nums.size())
                    break;
                if (dp[i+j] == -1) {
                    dp[i+j] = dp[i] + 1;
                } else {
                    dp[i+j] = std::min(dp[i+j], dp[i] + 1);
                }
            }
        }
        for (const auto &num : dp) {
            std::cout << num << std::endl;
        }
        return dp[nums.size()-1];
    }
};
